### DISCUSSION TITLE
Parsing JSON from multiple buffer calls.

### DISCUSSION THREAD

**Discussion by dawinaj** [nlohmann/json#4303] [Q&A]
I have an ESP32 with a webserver running. User sends some config in POST as JSON.
Now, these configs may(or may not) get quite long, so I would like to avoid copying the entire thing into a buffer and then parsing.
I guess I should make some kind of stream or iterator, which would feed characters into the parser, and then if internal buffer is empty, call a function to read some length of the request again. However I'm not sure how I can implement that, which `std` class to inherit from, etc.

The function declaration is `int httpd_req_recv(httpd_req_t *r, char *buf, size_t buf_len)`.
https://github.com/espressif/esp-idf/blob/master/components/esp_http_server/include/esp_http_server.h#L905

**Comment by nlohmann** [comment]
The parse function (https://json.nlohmann.me/api/basic_json/parse/) can read from a lot of inputs, including iterator ranges.

**Reply by dawinaj** [reply]
I would do it with two pointers, but I have no access to the internal memory address. If it even exists, because now I think it's being written straight from the socket.
I guess I would have to implement custom `istream`.
Unless, maybe I can treat socket as a `FILE*`? But then, there is no way to limit the length probably, and it will read past the POST field and maybe even break the whole API...
